.github/workflows/publish-new-release.yml

```diff
@@ -0,0 +1,191 @@
+name: "Publish new release"
+
+on:
+  pull_request:
+    branches:
+      - master
+    types:
+      - closed
+
+jobs:
+  build_binary:
+    if: github.event.pull_request.merged == true && startsWith(github.event.pull_request.head.ref, 'release/') # only merged release branches must trigger this
+    name: Build binary
+    strategy:
+      matrix:
+        os: [ubuntu, macos]
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+      - name: Checkout merge commit
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.merge_commit_sha }}
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          override: true
+
+      - name: Build ${{ matrix.os }} release binary
+        id: build
+        run: make build BUILD_ARGS='--release'
+
+      - name: Extract version from branch name
+        id: extract-version
+        run: |
+          BRANCH_NAME="${{ github.event.pull_request.head.ref }}"
+          VERSION=${BRANCH_NAME#release/}
+
+          echo "::set-output name=version::$VERSION"
+
+      - name: Create archive
+        id: create-archive
+        run: |
+          MACHINE=$(uname -m)
+          KERNEL=$(uname -s)
+          VERSION=${{ steps.extract-version.outputs.version }}
+
+          ARCHIVE="cnd_${VERSION}_${KERNEL}_${MACHINE}.tar.gz"
+
+          tar -C ./target/release --create --file=$ARCHIVE cnd
+
+          echo "::set-output name=archive::$ARCHIVE"
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.os }}-release-archive
+          path: ${{ steps.create-archive.outputs.archive }}
+
+  create_docker_image:
+    if: github.event.pull_request.merged == true && startsWith(github.event.pull_request.head.ref, 'release/') # only merged release branches must trigger this
+    name: Create and publish Docker image
+    needs: build_binary
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout merge commit
```

Comment:
Do you need to checkout the merge commit here? Further down you only download the release, extract it and build the container.

Reply:
I need the Dockerfile, which is in the repository :)